xhistogram/core.py

```diff
@@ -294,4 +367,4 @@ def reshape_input(a):
         final_shape = kept_axes_shape + h.shape[1:]
         h = reshape(h, final_shape)
 
-    return h
+    return h, bins
```

Comment:
This is a major API change that will break lots of existing code. If we consider this function part of the public API, then we should really do a deprecation cycle with warnings before changing this behavior.

I personally always use the xarray.histogram function rather than core.histogram. But maybe other users use this function directly?

Thoughts?

Reply:
I also only use xarray.histogram, and I imagine most xhistogram users are the same. But yes, this is a breaking change, so I agree a depreciation cycle is appropriate.

Otherwise, we could refactor so that all the bin logic is done within xarray.histogram, with core.histogram unchanged. The disadvantage of this is that then the xarray.histogram and core.histogram APIs are more different (i.e. all bin options would be available for xarray.histogram, but core.histogram would accept only numpy arrays as before). 

Happy to do either 🙂

Reply:
I will defer to your judgement on what to do here.

Reply:
I think it may be cleanest to keep what I have. This keeps the xarray.histogram and core.histogram APIs as similar as possible, and means that the core.histogram API overlaps strongly with the np.histogram API.

I'll submit a PR with the depreciation warning shortly.